### PR TITLE
feat(use-business): Implements use-business

### PR DESCRIPTION
**Pull request by grebaldi**
`useBusiness` provides a function `perform` that can be wrapped around a call to any process that returns a `Promise`. It also exposes a property `isBusy` that'll be `true` as long as any `Promise` that has been started is still running.

If `perform` gets called multiple times, `isBusy` stays `true` until all `Promises` are settled.

An example use case looks like this:
```typescript
const LoadMoreList = <I>(props: {
  loadItems: (page: number) => Promise<I[]>;
  renderItem: (item: I) => React.ReactNode;
}) => {
  const [items, setItems] = React.useState<I[]>([]);
  const [page, setPage] = React.useState(0);
  const business = useBusiness();
  const loadMore = Reac.useCallback(() => setPage((page) => page + 1), [
    setPage,
  ]);

  React.useEffect(async () => {
    const loadedItems = await business.perform(() => props.loadItems(page));

    setItems((items) => [...items, ...loadedItems]);
  }, [page]);

  return (
    <>
      <ul>
        {items.map((item, index) => (
          <li key={String(index)}>
            {renderItem(item)}
          </li>
        ))}
        {business.isBusy ? <li key="loading">Loading...</li> : null}
      </ul>
      <button onClick={loadMore} disabled={business.isBusy}>
        Load more
      </button>
    </>
  );
};
```

### CODE REVIEW

File: packages/use-business/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+```typescript
+const LoadMoreList = <I>(props: {
+  loadItems: (page: number) => Promise<I[]>;
+  renderItem: (item: I) => React.ReactNode;
+}) => {
+  const [items, setItems] = React.useState<I[]>([]);
+  const [page, setPage] = React.useState(0);
+  const business = useBusiness();
+  const loadMore = Reac.useCallback(() => setPage((page) => page + 1), [
```
Comment (henrikradandt) [inline packages/use-business/README.md:54]: Just a typo here.

```suggestion
  const loadMore = React.useCallback(() => setPage((page) => page + 1), [
```

File: packages/use-business/README.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    setPage,
+  ]);
+
+  React.useEffect(async () => {
+    const loadedItems = await props.loadItems(page);
+
+    setItems((items) => [...items, ...loadedItems]);
+  }, [page]);
```
Comment (henrikradandt) [inline packages/use-business/README.md:62]:
```suggestion
}, [page, setItems]);

File: packages/use-business/README.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+```typescript
+const LoadMoreList = <I>(props: {
+  loadItems: (page: number) => Promise<I[]>;
+  renderItem: (item: I) => React.ReactNode;
+}) => {
+  const [items, setItems] = React.useState<I[]>([]);
+  const [page, setPage] = React.useState(0);
+  const business = useBusiness();
+  const loadMore = Reac.useCallback(() => setPage((page) => page + 1), [
```
Comment (henrikradandt) [inline packages/use-business/README.md:95]: Same here.

```suggestion
  const loadMore = React.useCallback(() => setPage((page) => page + 1), [

File: packages/use-business/README.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    setPage,
+  ]);
+
+  React.useEffect(async () => {
+    const loadedItems = await business.perform(() => props.loadItems(page));
+
+    setItems((items) => [...items, ...loadedItems]);
+  }, [page]);
```
Comment (henrikradandt) [inline packages/use-business/README.md:103]: And here.

```suggestion
  }, [page, setItems]);

**Review by henrikradandt — APPROVED** [review]
Hi @grebaldi,

looks as good as useful and tests also work.
I just spotted a typo and a missing dependency in your README.md

**Comment by grebaldi** [comment]
Thanks a lot for your review @henrikradandt!

**Pull request merged**